kiwi/package_manager/base.py

```diff
@@ -182,3 +183,8 @@ def cleanup_requests(self):
         del self.package_requests[:]
         del self.collection_requests[:]
         del self.product_requests[:]
+
+    def __del__(self):
+        log.info('Cleaning up all repos from %s instance', type(self).__name__)
+        if self.repository is not None:
```

Comment:
Yes this Repository class that can handle multiple repositories used to confuse me a little also. I take your suggestion.